### PR TITLE
use async crypto.randomBytes

### PR DESCRIPTION
**Pull request by jcallu**

### CODE REVIEW

File: lib/auth/scram.js [hunk 1]
```diff
@@ -296,7 +294,10 @@ ScramSHA1.prototype.auth = function(server, connections, db, username, password,
 
   var _execute = function(_connection) {
     process.nextTick(function() {
-      executeScram(_connection);
+      crypto.randomBytes(24, function(eIgnore, buff) {
+        var nonce = buff.toString('base64');
```
Comment (mbroadst) [inline lib/auth/scram.js]: can you please throw an error here if it exists?

**Comment by mbroadst** [comment]
@jcallu thanks!

**Comment by jcallu** [comment]
@mbroadst was this merged to the base? can't find the change, if not. is there a plan?

**Comment by mbroadst** [comment]
@jcallu 😱 it seems your commit was lost in a rebranching of the project, do you still have the commit locally? Otherwise I can reprepare the code changes and merge them, but I want you to get attribution for the work. Many apologies for this!

**Pull request merged**